### PR TITLE
feat: add keyring:// URI scheme for custom-service credential lookup

### PR DESCRIPTION
**Pull request by nf-matt**
## Summary

This adds a new `keyring://service/account` URI scheme for `env_credentials` that lets users load secrets from the system keyring using a custom service name.

**Motivation:** Tools like `gh` store OAuth tokens as generic keychain passwords with their own service names (e.g., `gh:github.com`). The existing `apple-password://` scheme only supports internet passwords via `find-internet-password`, and the bare keyring fallback hardcodes the service name to `nono`. This fills that gap using the cross-platform `keyring` crate that's already a dependency.

### What's included

- **URI validation** — forbidden chars, structural checks, 1024-byte length limit
- **Account redaction** — `keyring://gh:github.com/alice` → `keyring://gh:github.com/<redacted>` in logs
- **Dispatch** in `load_secret_by_ref` via the existing `keyring` crate (`Entry::new(service, account)`)
- **List-mode rejection** — requires `--env-credential-map` (same as `apple-password://`)
- **Pair-mode validation** in `build_mappings_from_pairs`
- **Profile validation** in `validate_env_credential_keys`
- **`?decode=go-keyring`** — opt-in post-load decoding for Go tools that use `zalando/go-keyring` (e.g., `gh`), which wraps stored values with `go-keyring-base64:` prefix + base64 encoding
- **32 new tests** covering validation, redaction, decode transform, list/pair modes, and edge cases

### Usage

Profile:
```json
{
  "env_credentials": {
    "keyring://gh:github.com/alice": "GH_TOKEN"
  }
}
```

With Go-keyring decoding (for `gh` and other Go CLI tools):
```json
{
  "env_credentials": {
    "keyring://gh:github.com/alice?decode=go-keyring": "GH_TOKEN"
  }
}
```

CLI:
```bash
nono --env-credential-map 'keyring://gh:github.com/alice?decode=go-keyring' GH_TOKEN -- gh auth status
```

### Cross-platform

No `#[cfg]` gates needed — the `keyring` crate abstracts over macOS Keychain (generic passwords), Linux Secret Service (GNOME Keyring/KWallet), and Windows Credential Manager.

### Design decisions

- **Opt-in decoding** via query params rather than auto-detection — safer, extensible, and explicit about what transform is applied
- **Query param allowlist** — unknown params/values are rejected to prevent silent misconfiguration
- **No new dependencies** — base64 decoding uses the existing `trust::base64` module

### Files changed

- `crates/nono/src/keystore.rs` — URI scheme implementation + tests
- `crates/nono/src/lib.rs` — re-exports
- `crates/nono-cli/src/profile/mod.rs` — profile validation + tests

## Test plan

- [x] `make clippy` — clean
- [x] `make fmt-check` — clean
- [x] `make test` — all new tests pass (135 keystore, 11 profile)
- [x] Manual test: loaded `gh` OAuth token from macOS Keychain via `keyring://gh:github.com/<account>?decode=go-keyring` into a sandboxed nono session, then used the decoded token to create this PR

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: crates/nono/src/keystore.rs [hunk 1]
```diff
@@ -825,6 +1042,82 @@ fn load_from_apple_password(uri: &str) -> Result<Zeroizing<String>> {
...
+                    "failed to base64-decode go-keyring value for '{}': {}",
+                    redacted_uri, e
+                ))
+            })?;
+            let decoded = String::from_utf8(bytes).map_err(|_| {
+                NonoError::ConfigParse(format!(
+                    "go-keyring decoded value for '{}' is not valid UTF-8",
+                    redacted_uri
+                ))
+            })?;
```
[the remark this answers is not in the document]
Comment (nf-matt) [reply crates/nono/src/keystore.rs:1115]: Good observation — this is intentional. The `env_credentials` pipeline injects values as environment variables, which are inherently strings (`Zeroizing<String>` throughout). Binary secrets can't flow through this path regardless of the decode step.

The `ConfigParse` error here is consistent with the other decode failure cases in this function (missing prefix, invalid base64) — all are configuration/format errors where the user specified `?decode=go-keyring` but the stored value doesn't match expectations. A user hitting this would need to either remove the `?decode=` param or investigate what their tool actually stored.

If binary secret support were ever needed, it would require a separate `get_secret() -> Vec<u8>` path with a different injection mechanism (e.g., tmpfs file descriptor), which is well outside the scope of this URI scheme.

**Review by lukehinds — APPROVED** [review]
